Varadic macros are a C99 feature

`src/trace.h` uses varadic macros, which are a C99 feature:

```
$ gcc -std=c89 -Wall -pedantic test.c 
test.c:3:15: warning: anonymous variadic macros were introduced in C99 [-Wvariadic-macros]
    3 | #define MACRO(...) __VA_ARGS__
      |               ^~~
```

This should probably be changed to a wrapper function for non-C99 compilers.

--- a/src/trace.h
+++ b/src/trace.h
@@ -21,31 +21,53 @@ struct git_trace_data {
 
 extern struct git_trace_data git_trace__data;
 
-GIT_INLINE(void) git_trace__write_fmt(
+GIT_INLINE(void) git_trace__vwrite_fmt(
 	git_trace_level_t level,
-	const char *fmt, ...)
+	const char *fmt, va_list ap)
 {
 	git_trace_cb callback = git_trace__data.callback;
 	git_buf message = GIT_BUF_INIT;
-	va_list ap;
 
-	va_start(ap, fmt);
 	git_buf_vprintf(&message, fmt, ap);
-	va_end(ap);
 
 	callback(level, git_buf_cstr(&message));
 
 	git_buf_dispose(&message);
 }
 
+GIT_INLINE(void) git_trace__write_fmt(
+	git_trace_level_t level,
+	const char *fmt, ...)
+{
+	va_list ap;
+
+	va_start(ap, fmt);
+	git_trace__vwrite_fmt(level, fmt, ap);
+	va_end(ap);
+}
+
 #define git_trace_level()		(git_trace__data.level)
+/* Varadic macros are a C99 feature */
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
 #define git_trace(l, ...)		{ \
 									if (git_trace__data.level >= l && \
 										git_trace__data.callback != NULL) { \
 										git_trace__write_fmt(l, __VA_ARGS__); \
 									} \
 								}
+#else
+GIT_INLINE(void) git_trace(git_trace_level_t level, const char *fmt, ...)
+{
+	if (git_trace__data.level >= level &&
+			git_trace__data.callback != NULL) {
+		va_list ap;
 
+		va_start(ap, fmt);
+		git_trace__vwrite_fmt(level, fmt, ap);
+		va_end(ap);
+	}
+}
+#endif
 #else
 
 GIT_INLINE(void) git_trace__null(

--- a/src/trace.h
+++ b/src/trace.h
@@ -47,15 +47,6 @@ GIT_INLINE(void) git_trace__write_fmt(
 }
 
 #define git_trace_level()		(git_trace__data.level)
-/* Varadic macros are a C99 feature */
-#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
-#define git_trace(l, ...)		{ \
-									if (git_trace__data.level >= l && \
-										git_trace__data.callback != NULL) { \
-										git_trace__write_fmt(l, __VA_ARGS__); \
-									} \
-								}
-#else
 GIT_INLINE(void) git_trace(git_trace_level_t level, const char *fmt, ...)
 {
 	if (git_trace__data.level >= level &&
@@ -67,7 +58,6 @@ GIT_INLINE(void) git_trace(git_trace_level_t level, const char *fmt, ...)
 		va_end(ap);
 	}
 }
-#endif
 #else
 
 GIT_INLINE(void) git_trace__null(

